### PR TITLE
leaderindicator: Differentiate between leaders and guides

### PR DESCRIPTION
**Pull request by ls-**
It's been a while since these were added, but there's actually 2 types of leaders, guides and leaders, the former are leaders of parties formed via LFD/LFR. [In the default](https://github.com/Gethe/wow-ui-source/blob/live/Interface/FrameXML/PartyMemberFrame.lua#L231-L242) [UI they use](https://github.com/Gethe/wow-ui-source/blob/live/Interface/FrameXML/TargetFrame.lua#L131-L142) [different icons](https://github.com/Gethe/wow-ui-source/blob/live/Interface/FrameXML/PlayerFrame.lua#L97-L110). Also, those `UnitIn*` checks are unnecessary.

### CODE REVIEW

File: .luacheckrc [hunk 1]
```diff
@@ -69,6 +69,7 @@ read_globals = {
 	'GetUnitPowerBarInfo',
 	'GetUnitPowerBarInfoByID',
 	'GetUnitPowerBarStringsByID',
+	'HasLFGRestrictions',
```
Comment (Rainrider) [inline .luacheckrc:72]: Is this wrongly indented or is it the damn GitHub mobile app?

**Review by Rainrider — COMMENTED** [review]
Not sure if it is the rendering of the diff in the app, sorry if so

File: .luacheckrc (hunk 1 above)
Comment (ls-) [reply .luacheckrc:72]: image
All tabs.

**Review by Rainrider — APPROVED** [review]

**Comment by ls-** [comment]
This PR went full circle 😁🔫

**Comment by Rainrider** [comment]
@ls- do you want to remove default textures next expac or why the compat mode thingy?

**Comment by ls-** [comment]
@Rainrider I originally wanted to push this stuff during SL, so I added the compat mode thingy, but I decided to delay it until the next xpack, so I'll remove it later on.

**Comment by siweia** [comment]
Is it possible to merge LeaderIndicator and AssistantIndicator into one? There shouldn't be someone mark as both roles, right?

**Comment by ls-** [comment]
> Is it possible to merge LeaderIndicator and AssistantIndicator into one? There shouldn't be someone mark as both roles, right?

Given the new functionality of the leader indicator, I'm not sure about that. But I'll review it for the next major. We'll see.

**Pull request closed without merging**